core/src/test/scala/integration/kafka/api/RackAwareAutoTopicCreationTest.scala

```diff
@@ -37,29 +43,52 @@ class RackAwareAutoTopicCreationTest extends KafkaServerTestHarness with RackAwa
 
   def generateConfigs =
     (0 until numServers) map { node =>
-      TestUtils.createBrokerConfig(node, zkConnect, enableControlledShutdown = false, rack = Some((node / 2).toString))
+      TestUtils.createBrokerConfig(node, zkConnectOrNull, enableControlledShutdown = false, rack = Some((node % 2).toString))
     } map (KafkaConfig.fromProps(_, overridingProps))
 
   private val topic = "topic"
 
-  @Test
-  def testAutoCreateTopic(): Unit = {
-    val producer = TestUtils.createProducer(bootstrapServers())
-    try {
-      // Send a message to auto-create the topic
-      val record = new ProducerRecord(topic, null, "key".getBytes, "value".getBytes)
-      assertEquals(0L, producer.send(record).get.offset, "Should have offset 0")
+  @ParameterizedTest(name = TestInfoUtils.TestWithParameterizedQuorumName)
+  @ValueSource(strings = Array("zk", "kraft"))
+  def testAutoCreateTopic(quorum: String): Unit = {
+    val props = new Properties()
+    props.put(CommonClientConfigs.BOOTSTRAP_SERVERS_CONFIG, bootstrapServers())
+
+    TestUtils.waitUntilTrue(
+      () => brokers.head.metadataCache.getAliveBrokers().size == numServers,
+      "Timed out waiting for all brokers to become active")
+
+    // Send a message to auto-create the topic
+    val record = new ProducerRecord(topic, null, "key".getBytes, "value".getBytes)
+    assertEquals(
+      0L,
+      TestUtils.resource(TestUtils.createProducer(bootstrapServers())) { producer =>
+        producer.send(record).get.offset
+      },
+      "Should have offset 0")
 
-      // double check that the topic is created with leader elected
-      TestUtils.waitUntilLeaderIsElectedOrChanged(zkClient, topic, 0)
-      val assignment = zkClient.getReplicaAssignmentForTopics(Set(topic)).map { case (topicPartition, replicas) =>
-        topicPartition.partition -> replicas
+    var topicDescription : TopicDescription = null
+    try {
+      topicDescription = TestUtils.resource(Admin.create(props)) { adminClient =>
+        adminClient.describeTopics(Collections.singleton(topic)).topicNameValues().get(topic).get()
       }
-      val brokerMetadatas = adminZkClient.getBrokerMetadatas(RackAwareMode.Enforced)
-      val expectedMap = Map(0 -> "0", 1 -> "0", 2 -> "1", 3 -> "1")
-      assertEquals(expectedMap, brokerMetadatas.map(b => b.id -> b.rack.get).toMap)
-      checkReplicaDistribution(assignment, expectedMap, numServers, numPartitions, replicationFactor)
-    } finally producer.close()
+    } catch {
+      case e: ExecutionException => fail(e.getCause)
+    }
+    assertTrue(topicDescription != null, "Topic 'topic' should exist")
```

Comment:
Is this check necessary? Is there any case where the admin client might return a null description?